crypto/bn/bn_gf2m.c

```diff
@@ -336,6 +336,8 @@ int BN_GF2m_mod_arr(BIGNUM *r, const BIGNUM *a, const int p[])
         z[j] = 0;
 
         for (k = 1; p[k] != 0; k++) {
+            if (p[k] < 0 || p[k] >= p[k - 1])
```

Comment:
so I'm not thrilled with this.  It's like passing NULL to string functions.  I suppose if there's a chance this could be called remotely, it's okay.  But I'd rather see a utility function that like ```GF2M_factors_okay``` or similar.  Which also error-returned on too many subscripts.  And then callers of the GF2M functions could call that routine, first.